src/maxdiffusion/generate_sdxl.py

```diff
@@ -132,14 +129,28 @@ def run(config):
     mesh=mesh
   )
 
+  # if this checkpoint was trained with maxdiffusion
+  # the training scheduler was saved with it, switch it
+  # to a Euler scheduler
+  if isinstance(pipeline.scheduler, FlaxDDPMScheduler):
+    noise_scheduler, noise_scheduler_state = FlaxEulerDiscreteScheduler.from_pretrained(
+      config.pretrained_model_name_or_path,
+      revision=config.revision, subfolder="scheduler", dtype=jnp.float32
+    )
+    pipeline.scheduler = noise_scheduler
+    params["scheduler"] = noise_scheduler_state
+
   if config.lightning_repo:
     pipeline, params = load_sdxllightning_unet(config, pipeline, params)
 
   scheduler_state = params.pop("scheduler")
   old_params = params
   params = jax.tree_util.tree_map(lambda x: x.astype(weight_dtype), old_params)
   params["scheduler"] = scheduler_state
-  delete_pytree(old_params)
```

Comment:
why remove `delete_pytree(old_params)`?

Reply:
delete pytree was originally being used when some weights are loaded from pt, but now this logic is changed to keep pt -> jax weights in cpu before being moved to host and deleting the pytree is not needed anymore.